content/blog/2024/11/how-to-create-an-esp-idf-component/index.md

```diff
@@ -0,0 +1,483 @@
+        if(err != ESP_OK) {
+            ESP_LOGE(TAG, "Failed to read data from SHTC3 sensor");
+        } else {
+            ESP_LOGI(TAG, "Temperature: %.2f C, Humidity: %.2f %%", temperature, humidity);
+        }
+        vTaskDelay(1000 / portTICK_PERIOD_MS);
+    }
+}
+```
+
+On this task, the sensor will be woken up, the readout will be read and processed, and then the sensor will return to sleep mode.
+
+#### Creating Kconfig (optional)
+
+As a bonus, this component has a dedicated configuration menu using a `Kconfig` file. On this configuration menu, you will be able to set the I2C GPIOs (SDA and SCL), the I2C port number and the bus frequency.
+
+```text
+menu "Driver SHTC3 Sensor"
+            
+    menu "I2C"
+        config SHTC3_I2C_NUM
+            int "I2C peripheral index"
+            default -1
+            range -1 3
+            help
+                For auto select I2C peripheral, set to -1.
+
+        config SHTC3_I2C_SDA
+            int "I2C SDA pin"
+            default 17
+            range 0 55
+            help
+                Set the I2C SDA pin for the data signal.
+
+        config SHTC3_I2C_SCL
+            int "I2C SCL pin"
+            default 18
+            range 0 55
+            help
+                Set the I2C SCL pin for the clock signal.
+
+        config SHTC3_I2C_CLK_SPEED_HZ
+            int "I2C clock speed (Hz)"
+            default 100000
+            range 10000 400000
+            help
+                Set the I2C clock speed in Hz.
+    endmenu
+  
+endmenu
+```
+
+The `Kconfig` file should be placed on the component root directory. To set the values, you can use the command:
+
+```bash
+idf.py menuconfig
+```
+
+Having a component with a clear documentation and at least one example, can make your component frustration free. Developers enjoy when it just works!
+
+### Running the application
+
+To run the application, run the command to `flash` and `monitor` using the `idf.py`. The application will print the temperature and humidity.
+
+{{< asciinema
+  key="component_shtc3"
+  idleTimeLimit="2"
+  speed="1.5"
+  poster="npt:0:09"
+>}}
+
+
+## Conclusion
+
+Writing a new component is highly recommended for improving the project architecture. This is a very powerful approach and can be used for many different types of component, not only for drivers.
+
+In a future article, this component will be added to the [Component Registry](https://components.espressif.com/) to be easily shared across the community.
+
+## Reference
+
+- [ESP-Registry](https://components.espressif.com/)
+- [ESP-Registry Documentation](https://docs.espressif.com/projects/idf-component-manager/en/latest/)
+- [Compote Documentation](https://docs.espressif.com/projects/idf-component-manager/en/latest/reference/compote_cli.html)
+- [Component Examples](https://github.com/espressif/esp-bsp/tree/master/components)
+- [My Components](https://github.com/pedrominatel/esp-components)
```

Comment:
I would suggest this link: https://components.espressif.com/components?q=ns%3Apedrominatel